{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "307184ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pymongo import MongoClient"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "dac1696c",
   "metadata": {},
   "outputs": [],
   "source": [
    "CLIENT = MongoClient(\"mongodb://localhost:27017\")\n",
    "\n",
    "db = CLIENT.tcDB\n",
    "\n",
    "collection = db[\"tcServices\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "96c78335",
   "metadata": {},
   "outputs": [],
   "source": [
    "data = collection.find({\"project_id\" : \"TIPL-3273\"}, {\n",
    "    \"_id\" : 0,\n",
    "    \"project_info.pi_name\": 1,\n",
    "    \"project_info.email\": 1,\n",
    "    \"project_info.institution\" : 1,\n",
    "    \"project_info.lab_dept\": 1,\n",
    "    \"service_info.service_name\" : 1,\n",
    "    \"service_info.platform\" : 1\n",
    "})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "332f0c5e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pymongo.synchronous.cursor.Cursor at 0x7db45585b890>"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0f1ede03",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[{'project_info': {'email': 'sriram.parameswaran@theracues.com',\n",
      "                   'institution': 'TheraCUES Innovations Pvt Ltd',\n",
      "                   'lab_dept': 'LAB',\n",
      "                   'pi_name': 'Sriram Parameswaran'},\n",
      "  'service_info': {'platform': 'Illumina',\n",
      "                   'service_name': 'Exome Sequencing - Standard (Germline)'}}]\n"
     ]
    }
   ],
   "source": [
    "from pprint import pprint\n",
    "\n",
    "dat = []\n",
    "\n",
    "for doc in data:\n",
    "    dat.append(doc)\n",
    "\n",
    "pprint(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "04ec5418",
   "metadata": {},
   "outputs": [],
   "source": [
    "categories = collection.find({}, {\"category\": 1, \"_id\": 0})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "42a2a703",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pymongo.synchronous.cursor.Cursor at 0x7d6ff378cbd0>"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "categories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "dda2b5c4",
   "metadata": {},
   "outputs": [],
   "source": [
    "product_classes = [cat['category'] for cat in categories]\n",
    "\n",
    "#for cat in categories:\n",
    "    #items = cat['category']\n",
    "    #product_classes.append(items)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "2c080613",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Bulk Transcriptomics',\n",
       " 'Metagenomics & Applied Genomics',\n",
       " 'Human Genetic Analysis',\n",
       " 'Applied',\n",
       " 'Targeted pathway interrogation (RUO)']"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "product_classes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "47664c92",
   "metadata": {},
   "outputs": [],
   "source": [
    "cat = \"Bulk Transcriptomics\"\n",
    "\n",
    "items = collection.find({\"category\" : cat}, {\"services.service_name\" : 1,\n",
    "                                              \"_id\" : 0,\n",
    "                                                \"services.application\": 1,\n",
    "                                                 \"services.supported_sample_types\" : 1,\n",
    "                                                 \"services.instrumentation\" : 1,\n",
    "                                                 \"services.process_map\" : 1,\n",
    "                                                 \"standard_deliverables\" : 1\n",
    "                                                   })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "da2801f9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[{'instrumentation': {'platform': 'Illumina',\n",
      "                      'read_length': '150 x 2 (Paired-end)'},\n",
      "  'process_map': ['Sample Extraction',\n",
      "                  'Sample QC',\n",
      "                  'Library Preparation and QC',\n",
      "                  'NGS Data Acquisition',\n",
      "                  'Primary Data QC',\n",
      "                  'Reference Alignment',\n",
      "                  'Sample Outlier Analysis',\n",
      "                  'Differential Expression Analysis (DEG)',\n",
      "                  'Pathway and Biological Interpretation'],\n",
      "  'service_name': 'RNA-seq (Eukaryotic Poly-A | With Reference Genome)',\n",
      "  'supported_sample_types': ['Eukaryotic cells',\n",
      "                             'Tissue',\n",
      "                             'PBMC',\n",
      "                             'Cancer tissue']},\n",
      " {'instrumentation': {'platform': 'Illumina',\n",
      "                      'read_length': '150 x 2 (Paired-end)'},\n",
      "  'process_map': ['Sample Extraction',\n",
      "                  'Sample QC',\n",
      "                  'Library Preparation and QC',\n",
      "                  'NGS Data Acquisition',\n",
      "                  'Primary Data QC',\n",
      "                  'De novo Transcriptome Assembly',\n",
      "                  'Annotation',\n",
      "                  'Reference Mapping',\n",
      "                  'Differential Expression Analysis (DEG)'],\n",
      "  'service_name': 'RNA-seq (Eukaryotic Poly-A | Non-Model Organism - De Novo)',\n",
      "  'supported_sample_types': ['Eukaryotic cells',\n",
      "                             'Plant tissue',\n",
      "                             'Fungal cells']},\n",
      " {'instrumentation': {'platform': 'Illumina',\n",
      "                      'read_length': '150 x 2 (Paired-end)'},\n",
      "  'process_map': ['RNA Extraction',\n",
      "                  'RNA QC',\n",
      "                  'rRNA Depletion',\n",
      "                  'Library Preparation and QC',\n",
      "                  'NGS Data Acquisition',\n",
      "                  'Reference Alignment',\n",
      "                  'Gene, Transcript, lncRNA and Other RNA Quantification',\n",
      "                  'Differential Expression Analysis (DEG)',\n",
      "                  'Biological Insights'],\n",
      "  'service_name': 'RNA-seq (Total RNA | Human / Mouse / Rat)',\n",
      "  'supported_sample_types': ['Tissue', 'Cancer tissue', 'PBMC', 'Serum']},\n",
      " {'instrumentation': {'platform': 'Illumina',\n",
      "                      'read_length': '150 x 2 (Paired-end)'},\n",
      "  'process_map': ['Sample Extraction',\n",
      "                  'Sample QC',\n",
      "                  'Ribosomal RNA Depletion',\n",
      "                  'Library Preparation',\n",
      "                  'NGS Data Acquisition',\n",
      "                  'Primary Data QC',\n",
      "                  'Reference Mapping and Annotation',\n",
      "                  'Differential Expression Analysis (DEG)',\n",
      "                  'Biological Insights'],\n",
      "  'service_name': 'RNA-seq (Bacterial | With Reference Genome)',\n",
      "  'supported_sample_types': ['Frozen samples',\n",
      "                             'RNA later preserved samples',\n",
      "                             'Inactivated bacterial samples']},\n",
      " {'instrumentation': {'platform': 'Illumina',\n",
      "                      'read_length': '150 x 2 (Paired-end)'},\n",
      "  'process_map': ['Whole Blood RNA Extraction',\n",
      "                  'RNA QC',\n",
      "                  'Globin Depletion',\n",
      "                  'Library Preparation and QC',\n",
      "                  'NGS Data Acquisition',\n",
      "                  'Primary Data QC',\n",
      "                  'Reference Mapping',\n",
      "                  'Differential Expression Analysis (DEG)',\n",
      "                  'Biological Insights'],\n",
      "  'service_name': \"RNA-seq (Blood | 3' End Tagging)\",\n",
      "  'supported_sample_types': ['Whole blood',\n",
      "                             'Stabilized blood (PAXgene, Tempus, or similar '\n",
      "                             'tubes)']},\n",
      " {'instrumentation': {'platform': 'Illumina',\n",
      "                      'read_length': '150 x 2 (Paired-end)'},\n",
      "  'process_map': ['FFPE Curl Preparation',\n",
      "                  'RNA Extraction',\n",
      "                  'RNA QC',\n",
      "                  'Ribosomal RNA Depletion and Directional Library Preparation',\n",
      "                  'NGS Data Acquisition',\n",
      "                  'Primary Data QC',\n",
      "                  'Reference Alignment',\n",
      "                  'Differential Expression Analysis (DEG)',\n",
      "                  'Biological Insights'],\n",
      "  'service_name': 'RNA-seq (FFPE | Human / Mouse / Rat | With Reference '\n",
      "                  'Genome)',\n",
      "  'supported_sample_types': ['FFPE tissue sections',\n",
      "                             'Archived clinical biopsy samples',\n",
      "                             'Tumor samples']},\n",
      " {'instrumentation': {'platform': 'Illumina',\n",
      "                      'read_length': '150 x 2 (Paired-end)'},\n",
      "  'process_map': ['FFPE Curl Preparation',\n",
      "                  'RNA Extraction',\n",
      "                  'RNA QC',\n",
      "                  'Multi-library RNA-seq for NGS',\n",
      "                  'NGS Data Acquisition (Illumina)',\n",
      "                  'Fragmented RNA Analysis Pipeline',\n",
      "                  'Gene Expression and Isoform Analysis',\n",
      "                  'Fusion Detection',\n",
      "                  'Tumor Signature Estimation',\n",
      "                  'Immune Profiling',\n",
      "                  'Differential Expression Analysis (DEG)'],\n",
      "  'service_name': 'FFPE Cancer Transcriptome (Discovery RNA-seq | Human)',\n",
      "  'supported_sample_types': ['FFPE tissue sections',\n",
      "                             'Archived clinical biopsy samples',\n",
      "                             'Tumor samples']},\n",
      " {'instrumentation': {'platform': 'Illumina',\n",
      "                      'read_length': '150 x 2 (Paired-end)'},\n",
      "  'process_map': {'bacterial_host_workflow': ['Total RNA Extraction',\n",
      "                                              'RNA QC',\n",
      "                                              'Ribosomal RNA Depletion',\n",
      "                                              'Library Preparation and QC',\n",
      "                                              'NGS Data Acquisition',\n",
      "                                              'Reference Alignment',\n",
      "                                              'De novo Annotation',\n",
      "                                              'Differential Expression '\n",
      "                                              'Analysis (DEG)',\n",
      "                                              'Biological Insights'],\n",
      "                  'fungal_host_workflow': ['Total RNA Extraction',\n",
      "                                           'RNA QC',\n",
      "                                           'Poly(A) Enrichment',\n",
      "                                           'Library Preparation and QC',\n",
      "                                           'NGS Data Acquisition',\n",
      "                                           'Primary Data QC',\n",
      "                                           'Host Reference (De novo Assembly '\n",
      "                                           'and Annotation)',\n",
      "                                           'Pathogen Reference Mapping',\n",
      "                                           'Differential Expression Analysis '\n",
      "                                           '(DEG)',\n",
      "                                           'Biological Insights']},\n",
      "  'service_name': 'RNA-seq (Plant / Microbial | Host-Pathogen Co-expression)',\n",
      "  'supported_sample_types': ['Infected plant tissues (leaf, root, stem)',\n",
      "                             'Plant-fungal interaction samples',\n",
      "                             'Plant-bacterial interaction samples',\n",
      "                             'Mixed host-pathogen RNA preparations']},\n",
      " {'instrumentation': {'platform': 'Illumina',\n",
      "                      'read_length': '150 x 2 (Paired-end)'},\n",
      "  'process_map': ['RNA Extraction',\n",
      "                  'RNA QC',\n",
      "                  'Library Preparation and QC',\n",
      "                  'NGS Data Acquisition',\n",
      "                  'Primary Data QC',\n",
      "                  'De novo Transcriptome Assembly',\n",
      "                  'Transcriptome Annotation',\n",
      "                  'Differential Expression Analysis (DEG)',\n",
      "                  'Biological Insights'],\n",
      "  'service_name': 'RNA-seq (Plant / Microbial | De Novo Transcriptome)',\n",
      "  'supported_sample_types': ['Plant tissues (leaf, root, stem)',\n",
      "                             'Fungal samples',\n",
      "                             'Bacterial cultures',\n",
      "                             'Algal samples']}]\n"
     ]
    }
   ],
   "source": [
    "from pprint import pprint\n",
    "for doc in items:\n",
    "    pprint(doc[\"services\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "79aea8e3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "fb2f4c70",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'RNA-seq (Eukaryotic Poly-A | With Reference Genome)'\n",
      "'RNA-seq (Eukaryotic Poly-A | Non-Model Organism - De Novo)'\n",
      "'RNA-seq (Total RNA | Human / Mouse / Rat)'\n",
      "'RNA-seq (Bacterial | With Reference Genome)'\n",
      "\"RNA-seq (Blood | 3' End Tagging)\"\n",
      "'RNA-seq (FFPE | Human / Mouse / Rat | With Reference Genome)'\n",
      "'FFPE Cancer Transcriptome (Discovery RNA-seq | Human)'\n",
      "'RNA-seq (Plant / Microbial | Host-Pathogen Co-expression)'\n",
      "'RNA-seq (Plant / Microbial | De Novo Transcriptome)'\n"
     ]
    }
   ],
   "source": [
    "from pprint import pprint\n",
    "\n",
    "cat = \"Bulk Transcriptomics\"\n",
    "\n",
    "items = collection.find({\"category\" : cat}, {\"services.service_name\" : 1, \"_id\" : 0})\n",
    "\n",
    "\n",
    "\n",
    "for doc in items:\n",
    "    for service in doc.get(\"services\", []):\n",
    "        pprint(service.get(\"service_name\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9959aba9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "437cc5a1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "ai-dev",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
